### PR TITLE
Issue #109 Workflow Script

### LINKED ISSUE #109: Automate Release Workflow to Build Desktop Application Executable For Windows.
Description:
Now that we have a bash script to create a BubbleScan .exe file, the next step is to automate this process. We need a CI/CD workflow that triggers the script and generates a release build every time a pull request is merged into the main branch. This will streamline our deployment and testing process, making it easier to distribute the latest version of BubbleScan as a desktop application.

Tasks:

- Set up a CI/CD workflow (using GitHub Actions) that triggers on merges to the main branch.
- Integrate the existing bash script (Bubblescan.sh) to generate the executable (Bubblescan.exe).
- Store the generated .exe file as a release asset in GitHub so that it’s accessible for download.

Workflow Steps:

- Trigger the workflow automatically on every merge to the main branch.
- Use a runner environment compatible with the bash script and PyInstaller.
- Run chmod +x Bubblescan.sh to make the script executable and execute ./Bubblescan.sh.
- Upload the generated Bubblescan.exe file to GitHub Releases with a versioned tag.

Success Criteria:

- [x] After every merge to main, a new release with the .exe file is automatically created and available for download in GitHub Releases.
- [x] The release version is updated and tagged appropriately.
- [x] The .exe file runs the application smoothly on supported operating systems and includes all necessary dependencies.

### PR DESCRIPTION
**Pull request by mylescorrea**
Fixes #109 

**What was changed?**

*I created a workflow script in .github/workflows/release.yml. I also changed the BubbleScan.sh file to now output a .exe file called 'BubbleScan' instead of AppServer*

**Why was it changed?**

*We have to automate the executable process so that if something new is pushed into the main branch, it's easier to release the latest version. *

**Just commit and push something on the branch then wait for 'Bubblescan release' in actions tab to complete**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        run: ./BubbleScan.sh
+
+      - name: Archive executable and static files
+        run: zip -r ServerCode/dist/Bubblescan.zip ServerCode/dist/BubbleScan.exe ServerCode/dist/static
+
+      - name: Create a new Git tag
+        id: create_tag
+        run: |
+          TAG="v$(date +'%Y%m%d%H%M%S')"
+          echo "TAG=$TAG" >> $GITHUB_ENV
```
Comment (ramezmosad) [inline .github/workflows/release.yml]: I'm not too familiar with yml files, but I do know that your code is supposed to generate a BubbleScan.exe, and automatically update it whenever new changes to code are made. I don't see the executable right now, but I'm sure you got this!

**Review by ramezmosad — COMMENTED** [review]
Great structure for the yaml file, I can't wait til it's fully implemented!

**Comment by kate-holdener** [comment]
@mylescorrea I saw we now have files in the "Releases" link of our repo. That's exciting. When I download BubbleScan.zip, I get a windows application. Do we have a mac version?

**Comment by mylescorrea** [comment]
> @mylescorrea I saw we now have files in the "Releases" link of our repo. That's exciting. When I download BubbleScan.zip, I get a windows application. Do we have a mac version?

Just fixed this right now

File: .github/workflows/release.yml
Comment (Chirag2x) [inline .github/workflows/release.yml:1]: The lml file covers all the bases.

File: BubbleScan.sh [hunk 2]
```diff
@@ -32,13 +32,38 @@ pip uninstall -y pathlib 2>/dev/null || echo "pathlib was not found, skipping."
...
+    echo "Building macOS binary..."
+    pyinstaller --onefile --name BubbleScan-macOS --add-data "application/static:static" --hidden-import=flask --hidden-import=werkzeug --hidden-import=fitz application/AppServer.py
+
+    echo "Building Windows binary..."
+    pyinstaller --onefile --name BubbleScan-Windows.exe --add-data "application/static;static" --hidden-import=fitz application/AppServer.py
+else
+    # Windows build
+    echo "Building Windows binary..."
+    pyinstaller --onefile --name BubbleScan-Windows.exe --add-data "application/static;static" --hidden-import=fitz application/AppServer.py
+fi
```
Comment (Chirag2x) [inline BubbleScan.sh:66]: Good Job here on identifying the Operating System.

**Review by Chirag2x — APPROVED** [review]
All the things are working as it should be. Good Job here. Try to add comments on the code for understanding. I also had hard time doing that.

**Pull request merged**